examples/connext_dds/dynamic_data_request_reply/cpp/PrimeNumberReplier.cxx

```diff
@@ -0,0 +1,328 @@
+        replier_params.reply_type_support(reply_type_support);
+
+        DDS_DynamicData *reply = reply_type_support->create_data();
+         
+        /* In this example we create the replier on the stack, but you
+         * can create on the heap as well
+         */
+        connext::Replier<DDS_DynamicData, DDS_DynamicData> replier(
+                replier_params);
+
+        /*
+         * Receive requests and process them
+         */
+
+        const DDS::Duration_t MAX_WAIT = {20, 0};
+
+        /* receive_request returns false when it times out */
+        connext::LoanedSamples<DDS_DynamicData>received_requests = 
+                replier.receive_requests(MAX_WAIT);
+
+        bool stop=1;
+
+        while(stop) {
+
+            typedef connext::LoanedSamples<DDS_DynamicData>::iterator iterator;
+            for (
+                    iterator request_sample = received_requests.begin(); 
+                    request_sample != received_requests.end(); 
+                    ++request_sample) 
+            {
+                
+                if(!request_sample->info().valid_data) {
+                    continue;
+                }
+                
+                /* We obtain "n" and "primes_per_reply" 
+                 * from the received sample:
+                 */
+                
+                DDS_Long n, primes_per_reply; 
+
+                request_sample->data().get_long(
+                        n,
+                        "n",
+                        DDS_DYNAMIC_DATA_MEMBER_ID_UNSPECIFIED);
+
+                request_sample->data().get_long(
+                        primes_per_reply,
+                        "primes_per_reply",
+                        DDS_DYNAMIC_DATA_MEMBER_ID_UNSPECIFIED);
+
+                /* This constant is defined in PrimesType.cxx */
+                if (n <= 0 || primes_per_reply <= 0 
+                        || primes_per_reply > THE_PRIME_SEQUENCE_MAX_LENGTH)
+                {
+
+                    std::cout << "Cannot process request" << std::endl;
+
+                    /* Send reply indicating error */
+                    reply->set_long(
+                            "status",
+                            DDS_DYNAMIC_DATA_MEMBER_ID_UNSPECIFIED,
+                            2);
+
+                    replier.send_reply(
+                            *reply, 
+                            request_sample->identity());
+                }
+                else{
+                    std::cout << "Calculating prime numbers below "
+                          << n << "... " << std::endl;
+                }
+
+                calculate_and_send_primes(
+                        replier,
+                        request_sample->identity(),
+                        n,
+                        primes_per_reply,
+                        reply);
+
+                std::cout << "DONE" << std::endl;
+ 
+            }
+
+            stop=0;
```

Comment:
```suggestion
            stop = false;
```

Reply:
do we really need to control the loop with this boolean? Because whenever the for loop is finished, this will be false and the while loop finishes as well, right? Am I missing something?